src/testutils/blockchain_tests.rs

```diff
@@ -358,1112 +359,1611 @@ where
     }
 }
 
-/// This macro runs blockchain tests against a `Blockchain` implementation. It requires access to a
-/// Bitcoin core wallet via RPC. At the moment you have to dig into the code yourself and look at
-/// the setup required to run the tests yourself.
-#[macro_export]
-macro_rules! bdk_blockchain_tests {
-    (
-     fn $_fn_name:ident ( $( $test_client:ident : &TestClient )? $(,)? ) -> $blockchain:ty $block:block) => {
-        #[cfg(test)]
-        mod bdk_blockchain_tests {
-            use $crate::bitcoin::{Transaction, Network};
-            use $crate::testutils::blockchain_tests::TestClient;
-            use $crate::blockchain::Blockchain;
-            use $crate::database::MemoryDatabase;
-            use $crate::types::KeychainKind;
-            use $crate::wallet::AddressIndex;
-            use $crate::{Wallet, FeeRate, SyncOptions};
-            use $crate::testutils;
-
-            use super::*;
-
-            #[allow(unused_variables)]
-            fn get_blockchain(test_client: &TestClient) -> $blockchain {
-                $( let $test_client = test_client; )?
-                $block
+// Marker enum to direct testClient to AnyBlockChain conversion
+pub enum BlockchainType {
```

Comment:
This is a step backward compared to what we had before: the goal of the macro was not only to test our internal blockchains, but also to allow our users to test their blockchains against a set of edge cases that we maintain and that should hopefully be pretty complete.

This can also be done without a macro, but it needs to work with generics instead of `AnyBlockchain`, because `AnyBlockchain` cannot be extended outside of BDK (essentially it only contains our blockchain types).

Alternatively, instead of using the generics you might be able to use the [`impl Trait`](https://doc.rust-lang.org/reference/types/impl-trait.html), but we definitely need something more generic than this for our external users.

Reply:
Yes this makes sense.. I will work on it and update accordingly..